From 37d40d0f6135e4209bdc448c26eacd45f05d8ce4 Mon Sep 17 00:00:00 2001
From: Roberto Granados <x0095451@ti.com>
Date: Fri, 4 Mar 2011 19:03:03 -0600
Subject: OMAP4: HDMI: hdmi_lib_init should not depend on DSS clocks

We cannot assume the DSS clocks will be enabled in the case
hdmi_lib_init() gets called from the audio driver first.
CSR/DR: OMAPS00230459/OMAPS00231115

Change-Id: I5d2f7a897fcd11e9dfd247b7f8691ccfd6635df8
Signed-off-by: Roberto Granados Dorado <x0095451@ti.com>
---
 arch/arm/plat-omap/hdmi_lib.c |   15 +++++++++------
 1 files changed, 9 insertions(+), 6 deletions(-)

diff --git a/arch/arm/plat-omap/hdmi_lib.c b/arch/arm/plat-omap/hdmi_lib.c
index 5ddbc40..ca3a7c7 100644
--- a/arch/arm/plat-omap/hdmi_lib.c
+++ b/arch/arm/plat-omap/hdmi_lib.c
@@ -1663,8 +1663,15 @@ int hdmi_lib_enable(struct hdmi_config *cfg)
 	return r;
 }
 
-int hdmi_lib_init(void){
-	u32 rev;
+/**
+ * hdmi_lib_init - Initializes hdmi library
+ *
+ * hdmi_lib_init is expected to be called by any user of the hdmi library (e.g.
+ * HDMI video driver, HDMI audio driver). This means that it is not safe to add
+ * anything in this function that requires the DSS clocks to be enabled.
+ */
+int hdmi_lib_init(void)
+{
 	static u8 initialized;
 
 	mutex_lock(&hdmi_mutex);
@@ -1693,10 +1700,6 @@ int hdmi_lib_init(void){
 
 	INIT_LIST_HEAD(&hdmi.notifier_head);
 
-	rev = hdmi_read_reg(HDMI_WP, HDMI_WP_REVISION);
-	printk(KERN_INFO "OMAP HDMI W1 rev %d.%d\n",
-		FLD_GET(rev, 10, 8), FLD_GET(rev, 5, 0));
-
 	return 0;
 }
 
-- 
1.6.0.4

